models/users-models.js

```diff
@@ -0,0 +1,14 @@
+const db = require("../db/connection");
+
+function checkIfUserExists(username) {
+  return db
+    .query(`SELECT * FROM users WHERE username = $1`, [username])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "username not found" });
+      }
+      return rows;
+    });
+}
```

Comment:
It isn't be necessary to check the user exists. In the comments table, comments.author _must_ reference an existing user from the users table. If this is not the case, PSQL will throw an error for you to catch, It is more impressive to show your understanding of PSQL errors rather than handle errors yourself :)